### ISSUE TITLE
Stylelint does not work on CSS-in-JS when a .babelrc file is present

### ISSUE THREAD

**Issue by caribou-code** [stylelint/stylelint#5382]
> Clearly describe the bug
When a .babelrc file is present in a project and stylelint is run to lint CSS-in-JS it doesn't work at all.

> Which rule, if any, is the bug related to?
All of them

> What code is needed to reproduce the bug?
Adding stylelint to the [Next.js example for Styled Components](https://github.com/vercel/next.js/tree/master/examples/with-styled-components) should do it.

> What stylelint configuration is needed to reproduce the bug?
Any rules, but I'm currently using this:

```json
{
  "extends": "stylelint-config-recommended"
}
```

> Which version of stylelint are you using?

`13.13.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "**/*.ts"`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to CSS-in-JS

> What did you expect to happen?

Any relevant warnings to be flagged

> What actually happened (e.g. what warnings or errors did you get)?

No warnings are flagged. Stylelint doesn't work at all until the .babelrc file is deleted (however this would break the application).

> Additional notes

It looks like this problem has surfaced a few times in old issues. [Sometimes it's been flagged as fixed](https://github.com/stylelint/stylelint/issues/4018) and others it's been moved to [some kind of backlog](https://github.com/stylelint/stylelint/issues/4574) that hasn't been touched in 1.5 years.

**Comment by pcvthien** [comment]
To help narrow down the issue: for me the issue starts on version 13.3.1. 13.3.0 and lower work for me.

**Comment by jeddy3** [comment]
@caribou-code Thanks for the report and for using the template.

Unfortunately, the upstream package (`postcss-css-in-js`) has been intermittently maintained over the past couple of years since the original author became inactive on GitHub. 

>  Sometimes it's been flagged as fixed and others it's been moved to some kind of backlog that hasn't been touched in 1.5 years.

In the [next major release](https://github.com/stylelint/stylelint/issues/5205) we intend to [move away from monolithic syntaxes](https://github.com/stylelint/stylelint/issues/5289) in favour of smaller library-specific ones. Styled-components is a popular libary and a leaner syntax written for that will hopefully be more maintainable without the baggage of trying to support all the other CSS-in-JS flavours. 

Writing leaner syntaxes that don't monkey-patch PostCSS has only recently been made possible by the addition of a [`Document` node to PostCSS@8](https://github.com/postcss/postcss/releases/tag/8.3.0).

We need help getting version 14 ready, including writing reference syntaxes for a template literal library and one for an object-based one, so please consider [contributing](https://stylelint.io/contributing) if you have time.

**Comment by caribou-code** [comment]
@jeddy3 thanks for the update. Sorry I'm just coming back to this after a couple of months.

v14 sounds great and I'm looking forward to that! However the pressing issue right now is that Stylelint doesn't work at all with Styled Components, because a babelrc file is required. Due to the popularity of SC, that seems like a major problem and not something that should wait until the next major release. Is there anything that can be done as a patch? Or alternatively are there know Stylelint versions that don't have this bug?

**Comment by Lonli-Lokli** [comment]
But stylelint can with React & Styled-components, why next is a problem?
